### DISCUSSION TITLE
Should `packer_compiled.lua` be renamed to `__packer_compiled.lua` by default?

### DISCUSSION THREAD

**Discussion by mcchrish** [wbthomason/packer.nvim#565] [Ideas]
Since nvim source the files alphabetically we have to choose the default load order of the compiled file, it's either the very first or the very last. Having it in between can lead to confusion for new users. Adding some extra two `__` will at least guarantee it will be the last file to be sourced.

The argument for having it be loaded last by default is that some plugins rely on specific configurations (usually `vim.opt` and `vim.g`) to be set beforehand.

It's of course still possible to modify this behavior using custom `init` values.

**Comment by wbthomason** [comment]
Sorry for my delayed response; I was away due to work (#562). This is a reasonable point, but I think it may be better to leave things as they are for now - #402, which I hope to start working on in earnest this week or next, will change how the compiled information is loaded anyway, to load earlier in the Neovim startup sequence, which would obviate this change. I'll also note that the purpose of `packer`'s `setup` key is for code that needs to run *before* a plugin is loaded (e.g. your `vim.g`/`vim.opt` cases).
